lib/src/main/java/org/jvnet/localizer/ResourceProvider.java

```diff
@@ -0,0 +1,47 @@
+package org.jvnet.localizer;
+
+import java.net.URL;
+
+/**
+ * Allows customizing what class loader is used to find a given resource.
+ *
+ * By default, it's the class loader for the class of the generated resource bundle, but certain
+ * localization patterns might require a different source.
+ */
+public abstract class ResourceProvider {
```

Comment:
Why not interface? It would be more flexible, and AFAICT there is no need in abstract class per se.

Reply:
Interfaces cannot have `static` non-`final` fields.